### DISCUSSION TITLE
Poor UI performance with DirectX 11 on AMD system.

### DISCUSSION THREAD

**Discussion by RealistikDash** [ppy/osu#28251] [Q&A]
### Type

Performance

### Bug description

Following the latest update (cannot confirm its specifically the most recent update causing this), I am experiencing poor performance on the "beatmap listing" and "song select" screens on my AMD system. **This does not affect gameplay at all, which reaches the FPS cap of 1000.** I find it likely that the thing linking these two screens is beatmap images having to be scaled to fit a UI container, made worse by the scrolling functionality. **Switching to OpenGL fixes all of these issues.**

On the song select screen, the FPS when doing nothing is ~400 (already 2x lower than OpenGL). However, even when scrolling the smallest amount possible, the FPS drops to <80 for 2s, before returning to 400.

On beatmap listing, the performance is always poor unless there are very few results.

When this lag occurs, a few CPU cores can be observed maxing out:
![image](https://github.com/ppy/osu/assets/36131887/61232d8d-3478-4407-8978-823f93d3eebf)
Performing the same action with OpenGL enabled does not cause any core to be maxed out.

I do not have concrete numbers about how much higher the FPS was before the update, however prior I could not feel any drops before.

### Attempted Versions
I have tried versions as old as 4 months (2023.1114.1 and a few in-between) and even manually compiled a version of lazer with the latest version of osu!framework available on Nuget (2024.513.0). I am able to replicate this issue on all of these versions.

### Specs
GPU: AMD Radeon RX 7900XTX
CPU: AMD Ryzen 7 5800X
Game Drive: Samsung 980 PRO

### Screenshots or videos

Scrolling the slightest amount in song select drops the FPS to 80 for 2s:
https://github.com/ppy/osu/assets/36131887/0e3dec7b-032c-4d1a-ae8a-8286c58ef83e

Poor song browse performance (shows both high result count and low):
https://github.com/ppy/osu/assets/36131887/abc746a5-8c49-4b88-bccc-1127d1801687

### Version

2024.519.1

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/15370744/compressed-logs.zip)

**Comment by bdach** [comment]
cc @smoogipoo

**Comment by smoogipoo** [comment]
Could you test a previous release? The recent work I've done on graphics has not made it to the recent release.

**Reply by RealistikDash** [reply]
After extensive testing, I have concluded that the recent update is in fact not related to this issue (which is surprising as this is surely something I would have felt before??). I have tested releases as old as 4 months and even compiled Lazer with the latest available version of osu!framework available on Nuget and have always been able to replicate this issue.

As mentioned in a discussion below, the experimental DirectX 11 renderer **does not have this issue**.

**Comment by EVAST9919** [comment]
You said that OpenGL fixes your issues, but have you tried D3D (experimental)?

**Reply by RealistikDash** [reply]
Just tried the experimental D3D and indeed I am unable to replicate this issue.
